### PR TITLE
Removes tablet hard drives entirely (HDD & SSD)

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This completely removes HDD and SSD as computer parts.

Effects:
- Tablets now store programs on themselves, you can't move the programs over with a hard drive.
- Computers built with metal in-hand now should work from the get-go? Or at the very least, doesn't need a hard drive to get working. I will probably make the cpu a separate thing you print and insert into the PC as a replacement, but that will be in the future.
- Tablets are easier to build now since they have less parts in general, hopefully we can get that number to 0
- Data disks are now their own path, and still works as before. You can upload, download, move over, clone, anything to them. They're basically the only way to move programs from one place to another now.
- This includes the Master system's disk. It's a data disk now. With a lot of space. Beyond more than necessary.
- Virus disks still exist, they are still disks, but they have no space on them to do anything with it.

I also tried some attempts at code improvement where I thought it was necessary, and tried adding clearer comments in places I thought were lacking.

Also, comes with an UpdatePaths, don't worry about it!

## Why It's Good For The Game

Deals with half of https://github.com/tgstation/tgstation/issues/70692- the other half is in a separate PR.

HDD and SSD were the 2 worst parts of Tablets imo, and the only reason it took this long to remove was because I didn't know exactly how to manage it, and because I knew it would be a massive timesink.

We've been removing tablet parts in general, and this brings us one massive step closer to completing that goal, because computer parts has never been a good part of tablet-building. It's too confusing for any SS13 player to bother with, and has very little gain.
It's also beneficial for coders because there's a lot of annoyance around going through the tablet, then checking the part for a hard drive, check if it exists, then checking for the app on that. Instead, you can now just check if the tablet has an app. Much more simple on both ends.

## Changelog

:cl:
balance: Tablet hard drives no longer exist. Applications are now stored directly onto the tablet.
balance: Building a modular computer with metal now should be a proper functioning computer.
balance: The R&D's Master system hard drive is now a data disk, if ever you need one that desperately.
fix: Tablets (not Cyborg ones) now have pens again, like they had before.
/:cl:

### CODE REVIEW

**Comment by timothymtorres** [comment]
What about laptops and other computers?  I still miss PDA cartridges, they were simple to use.

**Comment by JohnFulpWillard** [comment]
> What about laptops and other computers?

They're the same, outside of my mention that building computers with 10 metal doesn't require a drive to work now.
Computer files can only be moved with data disks, instead of pulling a hard drive out and moving it into a different computer.

> I still miss PDA cartridges, they were simple to use.

we still have job data disks that have applications meant for certain jobs on them, you can clone them to your computer. It's basically cartridges without being able to run it off the disk (regardless of whether the app can even RUN on the hardware or not)

**Comment by Striders13** [comment]
does it mean you can't install arcade games into a borg anymore

**Comment by Striders13** [comment]
> You can, same as before, with a data disk,

do you have games on your borg?

File: _maps/shuttles/starfury_corvette.dmm [hunk 1]
```diff
@@ -120,6 +120,7 @@
 "ao" = (
 /obj/machinery/light/small/directional/west{
 	brightness = 3;
+	
 	},
```
Comment (ShizCalev) [inline _maps/shuttles/starfury_corvette.dmm:124]: No

File: _maps/shuttles/starfury_fighter1.dmm [hunk 2]
```diff
@@ -51,6 +51,7 @@
 /obj/item/multitool,
 /obj/machinery/light/small/directional/west{
 	brightness = 3;
+
```
Comment (ShizCalev) [inline _maps/shuttles/starfury_fighter1.dmm:54]: .

File: _maps/shuttles/starfury_corvette.dmm (hunk 1 above)
Comment (JohnFulpWillard) [reply _maps/shuttles/starfury_corvette.dmm:124]: this just happened on its own when i ran updatepaths i have no clue why

File: code/modules/modular_computers/computers/item/computer.dm [hunk 3]
```diff
@@ -158,9 +158,9 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 	QDEL_LIST(stored_files)
 	Remove_Messenger()
 
-	if(inserted_disk)
+	if(istype(inserted_disk))
```
Comment (Mothblocks) [inline code/modules/modular_computers/computers/item/computer.dm:161]: Why istype? When is this a non-disk non-null?

Comment (JohnFulpWillard) [reply code/modules/modular_computers/computers/item/computer.dm]: Destroy() is being called on tablets before Initialize() is, so they don't actually have a disk/pen yet.

image

**Comment by Assiente** [comment]
It looks like some kind of virus or malware. You need to remove it first, take care of a good antivirus. After that, you can use special data recovery software. May you be lucky.

**Comment by Mothblocks** [comment]
... Are you a bot

**Comment by ShizCalev** [comment]
> ... Are you a bot

Think so, probably picked up on the word virus in the op

**Comment by ShizCalev** [comment]
Oof, yeah deffo a bot.
image
image

File: code/modules/modular_computers/computers/item/tablet.dm [hunk 4]
```diff
@@ -340,44 +359,35 @@
...
 /obj/item/modular_computer/tablet/pda/update_overlays()
 	. = ..()
-	var/init_icon = initial(icon)
 	var/obj/item/computer_hardware/card_slot/card = all_components[MC_CARD]
-	if(!init_icon)
-		return
-	if(card)
-		if(card.stored_card)
-			. += mutable_appearance(init_icon, "id_overlay")
+	if(card && card.stored_card)
```
Comment (ShizCalev) [inline code/modules/modular_computers/computers/item/tablet.dm:381]:
```suggestion
	if(card?.stored_card)
```

**Comment by ShizCalev** [comment]
`code/modules/modular_computers/computers/item/computer.dm:415:error: MC_SDD: undefined var`

**Pull request merged**